File: types/offer.go
package types

import "time"

type OfferStatus string

const (
	OfferPending   OfferStatus = "pending"
	OfferAccepted  OfferStatus = "accepted"
	OfferRejected  OfferStatus = "rejected"
	OfferCanceled  OfferStatus = "canceled"
	OfferCompleted OfferStatus = "completed"
)

type Offer struct {
	ID          string      `json:"id"`
	UserID      string      `json:"user_id"`
	Product     Product     `json:"product"`
	Amount      int64       `json:"amount"`
	Status      OfferStatus `json:"status"`
	PickupNotes string      `json:"pickup_notes"`
	CreatedAt   time.Time   `json:"created_at"`
	UpdatedAt   time.Time   `json:"updated_at"`
}

File: web/src/types/offer.ts
import type { Product } from './product'

export type OfferStatus = 'pending' | 'accepted' | 'rejected' | 'canceled' | 'completed'
export interface Offer {
  id: string
  user_id: string
  product: Product
  amount: number
  comment: string
  status: OfferStatus
  created_at: string
  updated_at: string
}
